vector/src/main/java/im/vector/app/features/spaces/share/ShareSpaceBottomSheet.kt

```diff
@@ -0,0 +1,127 @@
+
+    override fun create(initialState: ShareSpaceViewState): ShareSpaceViewModel = viewModelFactory.create(initialState)
+
+    override fun injectWith(injector: ScreenComponent) {
+        injector.inject(this)
+    }
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): BottomSheetSpaceInviteBinding {
+        return BottomSheetSpaceInviteBinding.inflate(inflater, container, false)
+    }
+
+    override fun invalidate() = withState(viewModel) { state ->
+        super.invalidate()
+        val summary = state.spaceSummary.invoke()
+
+        val spaceName = summary?.name
+
+        if (state.postCreation) {
+            views.headerText.text = getString(R.string.invite_people_to_your_space)
+            views.descriptionText.setTextOrHide(getString(R.string.invite_people_to_your_space_desc, spaceName))
+        } else {
+            views.headerText.text = getString(R.string.invite_to_space, spaceName)
+            views.descriptionText.setTextOrHide(null)
+        }
+
+        views.inviteByMailButton.isVisible = false // not yet implemented
+        views.inviteByLinkButton.isVisible = state.canShareLink
+        views.inviteByMxidButton.isVisible = state.canInviteByMxId
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // XXX enable back when supported
+        views.inviteByMailButton.isVisible = false
+        views.inviteByMailButton.debouncedClicks {
+        }
+
+        views.inviteByMxidButton.debouncedClicks {
+            viewModel.handle(ShareSpaceAction.InviteByMxId)
+        }
+
+        views.inviteByLinkButton.debouncedClicks {
+            viewModel.handle(ShareSpaceAction.InviteByLink)
+        }
+
+        viewModel.observeViewEvents { event ->
+            when (event) {
+                is ShareSpaceViewEvents.NavigateToInviteUser -> {
+                    val intent = InviteUsersToRoomActivity.getIntent(requireContext(), event.spaceId)
+                    startActivity(intent)
+                }
+                is ShareSpaceViewEvents.ShowInviteByLing -> {
+                    startSharePlainTextIntent(
+                            fragment = this,
+                            activityResultLauncher = null,
+                            chooserTitle = getString(R.string.share_by_text),
+                            text = getString(R.string.share_space_link_message, event.spaceName, event.permalink),
+                            extraTitle = getString(R.string.share_space_link_message, event.spaceName, event.permalink)
+                    )
+                }
+            }
+        }
+    }
+
+    companion object {
+
+        fun show(fragmentManager: FragmentManager, spaceId: String, postCreation: Boolean = false): ShareSpaceBottomSheet {
+            return ShareSpaceBottomSheet().apply {
+                isCancelable = true
```

Comment:
We cannot cancel this Dialog?

Reply:
yes we can, i 'll remove as it's the default anyway

Reply:
ha yes, `true != false` :)